### PR TITLE
Move examples in a separate crate

### PR DESCRIPTION
**Pull request by zakorgy**
This change separates the examples from the WR crate, and removes unrelated dependencies from it.

In the future this can help the transition to new backends, like adding new features which should not exist in the WR crate.

Any thoughts? @kvark @gw3583

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2811)

### CODE REVIEW

**Comment by mrobinson** [comment]
@zakorgy Do you explaining a bit more how this helps in the development of new backends?

File: webrender/Cargo.toml [hunk 1]
```diff
@@ -63,10 +63,6 @@ optional = true
 
 [dev-dependencies]
 mozangle = "0.1"
```
Comment (kvark) [inline webrender/Cargo.toml]: why is `mozangle` left?

Comment (zakorgy) [reply webrender/Cargo.toml]: Because it is used by the `tests` folder not by the `examples`.

**Comment by kvark** [comment]
@mrobinson the motivation is to have "webrender" crate independent of the backend at the type level. I.e. wrench, examples, servo,etc need to choose the Metal/D3D12/Vulkan graphics backend with accompanying window integration crates (like `winit`), so it makes sense to have those dependencies moved out.

Examples is the only thing that's not a separate crate among the ones that need to actually create a window and initialize WR. Other projects, like gfx-rs or Amethyst, also move their examples out for this reason.

**Comment by mrobinson** [comment]
@kvark Thanks for the explanation.

**Comment by kvark** [comment]
There doesn't appear to be strong opposition to doing this, even though there isn't much support either. Let's get it moving.
@bors-servo r+

**Pull request merged**